### ISSUE TITLE
Add more activity types / "generic" activity

### ISSUE THREAD

**Issue by arminus** [dreeveapp/dreeve#62]
I don't know if this is on the roadmap or not, but it would be nice to have support for more activities.

One way of approaching this would be to import basically every activity and show it with some generic statistics, like time, distance (if provided), heartrate, etc. This would bring the data into the backend and then maybe make it easier to community develop PRs for individual activity-type specific statistics mid-term.

For my part, I'd be interested in Hiking, Skitouring and Cross-Country Skiing, all of which are kind of similar to running, at least initially. See also https://github.com/flopp/activities for instance which is not activity specific at all (but has less statistics).

**Comment by robiningelbrecht** [comment]
That's a good idea indeed. I'd need a generic sport type icon as well then

**Comment by arminus** [comment]
Even if the icon is generic, I would keep the activity as string though, so that a generic activity can then be filtered with https://github.com/robiningelbrecht/strava-statistics/issues/43 ?

**Comment by robiningelbrecht** [comment]
Yeah ofcourse, but the activity titles are prefixed with an activity type icon. I was referencing those

**Comment by robiningelbrecht** [comment]
@arminus I'm going to pick up this feature soon. Would you be willing to test this for me? :) 
I only have bike activities on my Strava...

**Comment by arminus** [comment]
Gladly, I have many different activity types.

**Comment by robiningelbrecht** [comment]
@arminus ok cool, thanks! I'll keep you up to date

**Comment by robiningelbrecht** [comment]
@arminus I think the first version for "all activity types" is ready. I made a separate Docker Image for you to test:

`robiningelbrecht/strava-statistics:ISSUE-62-bis`

I'll leave here what I will add to the release notes when releasing this major update. Please let me know if you find any critical issues. Please be aware that the updated readme is not merged yet :p

> v0.3.0 is here! This version adds support for all activity types. 
>
> ⚠️ This release introduces a breaking change, please read https://github.com/robiningelbrecht/strava-statistics/wiki/Migrate-from-v0.2.x-to-v0.3.x before upgrading

**Comment by robiningelbrecht** [comment]
@yroyathon are you up for testing this? I think you have other types of activities as well?

**Comment by yroyathon** [comment]
@robiningelbrecht I'm open to help, but I'm not sure I have interesting activity types.  AFAIK I just have bike rides and runs, unless indoor runs and indoor rides count as separate activity type.  I def have a bunch of treadmill runs, and I might have 2 or 3 indoor bike rides.

**Comment by yroyathon** [comment]
@robiningelbrecht Actually I'm helping my brother setup a dashboard for his activities as well, and he definitely has one activity type I don't have (swims), maybe more (hikes/etc.).  So maybe I can help after all.  How does this change what I put in the ACTIVITY_TYPES_TO_IMPORT in the .env file?

**Comment by yroyathon** [comment]
@robiningelbrecht I don't know how popular all of the dozens? of activity types are on Strava, but if you made a good amount of support for running/biking/swimming, ie the big 3 for triathlons, I bet a lot of folks would use that.

**Comment by robiningelbrecht** [comment]
@yroyathon I added support for ALL activity types ;)

I think the first version for "all activity types" is ready. I made a separate Docker Image for you to test:

`robiningelbrecht/strava-statistics:ISSUE-62-bis`

I'll leave here what I will add to the release notes when releasing this major update. Please let me know if you find any critical issues. Please be aware that the updated readme is not merged yet :p

> v0.3.0 is here! This version adds support for all activity types.
>
> ⚠️ This release introduces a breaking change, please read https://github.com/robiningelbrecht/strava-statistics/wiki/Migrate-from-v0.2.x-to-v0.3.x before upgrading

**Comment by yroyathon** [comment]
@robiningelbrecht Ok I'll eventually migrate my 2 dashboards over to this new image for testing, one is in the middle of a long initial import though.  Just curious, why the variable name change that introduces a breaking change?  Is it to match your code, or to match what Strava calls things?

**Comment by robiningelbrecht** [comment]
@yroyathon It's to match Strava's naming convention indeed. They have "Activity type" and "Sport type" and my code used them wrong, so I fixed this in this new big release

**Comment by yroyathon** [comment]
@robiningelbrecht Well it didn't break on my data/dashboard, but I only had 11 walks I think to import.  I looked through all the tabs/graphs/tables, and everything seemed ok.  No new X activity breakouts I should look for?  I didn't see any.  I suppose the first step is to import all the new activity type data, then new charts/views come later.  You gotta "Walk" before you can "AlpineSki", as the saying goes.

I'll try the new image/setting on my brother's richer dataset once the current import finishes, 600 activities imported so far, must be nearing the daily api limit.

**Comment by yroyathon** [comment]
@robiningelbrecht Ok I'm maybe halfway done the import on my brother's data, on the 2nd pass, and it failed during the import with an error I've never seen.  After the 1st pass import finished, I switched it over to use this new test image for all activities.  Here is the error:
In CurlFactory.php line 275:
  cURL error 6: Could not resolve host: dgtzuqphqg23d.cloudfront.net (see htt
  ps://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://dgtzuqphqg23d.
  cloudfront.net/HQSWHTNj3MRjDflOKEbekUqiaeJJq5I6NsXiQs32iqs-1536x2048.jpg

**Comment by yroyathon** [comment]
@robiningelbrecht I'm just going to re-run it for now as long as I have the api calls for it, in case the cloudfront thing was a one-off.

**Comment by robiningelbrecht** [comment]
@yroyathon 

> No new X activity breakouts I should look for? I didn't see any. I suppose the first step is to import all the new activity type data, then new charts/views come later.

Exactly ;)

> Ok I'm maybe halfway done the import on my brother's data, on the 2nd pass, and it failed during the import with an error I've never seen

I think that's a one time thing indeed, an error on Strava's end. Maybe they were in the middle of a deploy? dunno :)

> I'm just going to re-run it for now as long as I have the api calls for it, in case the cloudfront thing was a one-off.

Ok! Let me know if the new `v0.3.0` has any obvious breaking changes. If not, I'll release today.

Thanks for your time!!!

Edit: pushed a new version to `robiningelbrecht/strava-statistics:ISSUE-62-bis`. It contains a lot of improvements and features you asked for

**Comment by yroyathon** [comment]
@robiningelbrecht I'll check out the new image.  I'll still be importing for my brother's data, maybe even a few more days, so I probably won't have much to report on how the dashboard handles the new data until then.  

One question, didn't it used to be possible for the build html step to execute even with the import wasn't complete due to api limits?  I feel like way back when I started my own import, this was possible, but now it is not.  It was more usable then, because you could explore the dashboard in this initial phase, instead of having nothing to look at while you're being patient through days or even a week of strava importing.  Just curious about this change.

**Comment by robiningelbrecht** [comment]
@yroyathon this caused a lot of issues with gears not being imported yet and throwing errors for missing data. That's why I decided to only allow to render the dashboard when the import is complete. I don't want to be fixing errors for incomplete data

**Comment by yroyathon** [comment]
Gotcha, that makes sense to prevent all that.

**Comment by arminus** [comment]
@robiningelbrecht sorry, I'm a bit sick right now and not really up to testing. Will get back to this later...

**Comment by robiningelbrecht** [comment]
@arminus no problem! Hope you get well soon
